src/concurrency/local_epoch.cpp

```diff
@@ -82,35 +74,37 @@ namespace concurrency {
         epoch_map_.erase(epoch_ptr->epoch_id_);
         epoch_queue_.pop();
       } else {
-        PL_ASSERT(epoch_ptr->epoch_id_ > epoch_lower_bound_);
-        epoch_lower_bound_ = epoch_ptr->epoch_id_ - 1;
+        PL_ASSERT(epoch_ptr->epoch_id_ > epoch_id_lower_bound_);
+        epoch_id_lower_bound_ = epoch_ptr->epoch_id_ - 1;
         break;
       }
     }
 
     epoch_lock_.Unlock();
   }
 
-  uint64_t LocalEpoch::GetMaxCommittedEpochId(const uint64_t epoch_id) {
+  uint64_t LocalEpoch::GetExpiredEpochId(const uint64_t epoch_id) {
     epoch_lock_.Lock();
     // there's no epoch in this thread.
-    // which indicates that this thread is used or GC'd for some time.
+    // which indicates that this thread is never used or has been GC'd for some time.
     if (epoch_queue_.size() == 0) {
-      epoch_lower_bound_ = epoch_id - 1;
+      epoch_id_lower_bound_ = epoch_id - 1;
     } else {
       while (epoch_queue_.size() != 0) {
+        // get the smallest epoch.
         auto &epoch_ptr = epoch_queue_.top();
+        // if there's no transaction in the epoch
         if (epoch_ptr->txn_count_ == 0) {
```

Comment:
The global epoch manger should also be responsible for updating local epoch. Otherwise, if there's no transaction assigned to a certain thread, its corresponding epoch manager will never be updated.